### ISSUE TITLE
Potential bug? dice loss function

### ISSUE THREAD

**Issue by BartvanMarrewijk** [perrydoremi/PlantSegStudio#3]
For the DSPVformer the layer loss with self.fix_dice_loss_weight option is different then in the call_function. The mask_dice_loss is divided twice by the number of pred_masks.

InstanceCriterion __call__ function:
```python
            mask_bce_loss = torch.stack(mask_bce_losses).sum() / len(
                pred_masks)
            mask_dice_loss = torch.stack(mask_dice_losses).sum()

            if self.fix_dice_loss_weight:
                mask_dice_loss = mask_dice_loss / len(pred_masks) * 4

            if self.fix_mean_loss:
                mask_bce_loss = mask_bce_loss * len(pred_masks) \
                    / len(mask_bce_losses)
                mask_dice_loss = mask_dice_loss * len(pred_masks) \
                    / len(mask_dice_losses)
``` 

InstanceCriterion get_layer_loss function divided by **len(pred_masks) twice**:

```python
            mask_bce_loss = torch.stack(mask_bce_losses).sum() / len(
                pred_masks)
            mask_dice_loss = torch.stack(mask_dice_losses).sum() / len(
                pred_masks)

            if self.fix_dice_loss_weight:
                mask_dice_loss = mask_dice_loss / len(pred_masks) * 4

            if self.fix_mean_loss:
                mask_bce_loss = mask_bce_loss * len(pred_masks) \
                    / len(mask_bce_losses)
                mask_dice_loss = mask_dice_loss * len(pred_masks) \
                    / len(mask_dice_losses)
```

**Comment by perrydoremi** [comment]
Thanks, that's a sharp observation. PSS's InstanceCriterion is inherited unchanged (apart from formatting) from OneFormer3D ([instance_criterion.py](https://github.com/filaPro/oneformer3d/blob/main/oneformer3d/instance_criterion.py)), whose loss implementation in turn derives from SPFormer ([loss.py](https://github.com/sunjiahao1999/SPFormer/blob/main/spformer/model/loss.py)).

This behavior goes back to SPFormer: its final-layer forward() divides the BCE loss by the batch size but not the Dice loss, while get_layer_loss() divides both. OneFormer3D added fix_dice_loss_weight and applied the same block in get_layer_loss(), where the Dice loss was already averaged, so it is divided by the batch size twice there. 

I can't tell from the code whether the original behavior was intentional or part of a training-time trick.
